x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/action/GetInferenceFieldsActionRequestTests.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.action;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.TransportVersionUtils;
+import org.elasticsearch.xpack.core.inference.action.GetInferenceFieldsAction;
+import org.elasticsearch.xpack.core.ml.AbstractBWCWireSerializationTestCase;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import static org.elasticsearch.xpack.core.inference.action.GetInferenceFieldsAction.GET_INFERENCE_FIELDS_ACTION_TV;
+
+public class GetInferenceFieldsActionRequestTests extends AbstractBWCWireSerializationTestCase<GetInferenceFieldsAction.Request> {
+    @Override
+    protected Writeable.Reader<GetInferenceFieldsAction.Request> instanceReader() {
+        return GetInferenceFieldsAction.Request::new;
+    }
+
+    @Override
+    protected GetInferenceFieldsAction.Request createTestInstance() {
+        return new GetInferenceFieldsAction.Request(
+            randomIndentifierSet(),
+            randomFields(),
+            randomBoolean(),
+            randomBoolean(),
+            randomQuery(),
+            randomIndicesOptions()
+        );
+    }
+
+    @Override
+    protected GetInferenceFieldsAction.Request mutateInstance(GetInferenceFieldsAction.Request instance) throws IOException {
+        return switch (between(0, 5)) {
+            case 0 -> new GetInferenceFieldsAction.Request(
```

Comment:
nit: it's no big deal for this one as I wouldn't anticipate adding params to the request object frequently. However, I think following the pattern of assigning local variables for each arg, calling switch to mutate one of them, then calling the constructor once is cleaner and more idiomatic. Just for future reference, no need to change here.